src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -798,19 +1206,302 @@ export class ResultsViewPageStore {
+            return Promise.resolve(this.molecularData.result!.map(d=>{
+                let oncogenic = "";
+                if (molecularProfileIdToMolecularProfile[d.molecularProfileId].molecularAlterationType === "COPY_NUMBER_ALTERATION") {
+                    const oncoKbDatum = this.getOncoKbCnaAnnotation.result!(d);
+                    const datumOncogenic = ((oncoKbDatum && oncoKbDatum.oncogenic) || "").toLowerCase();
+                    if (ONCOKB_ONCOGENIC_LOWERCASE.indexOf(datumOncogenic) > -1) {
+                        oncogenic = datumOncogenic;
+                    }
+                }
+                return Object.assign({oncoKbOncogenic: oncogenic}, d);
+            }));
+        }
+    });
+
+    public getPutativeDriverInfo(mutation:Mutation) {
+        const oncoKbDatum:IndicatorQueryResp | undefined | null | false = this.mutationAnnotationSettings.oncoKb &&
+            this.getOncoKbMutationAnnotation.isComplete &&
+            this.getOncoKbMutationAnnotation.result(mutation);
+
+        let oncoKb:string = ((oncoKbDatum && oncoKbDatum.oncogenic) || "").toLowerCase();
+        if (ONCOKB_ONCOGENIC_LOWERCASE.indexOf(oncoKb) === -1) {
+            oncoKb = "";
+        }
+
+        const hotspots:boolean =
+            (this.mutationAnnotationSettings.hotspots &&
+            this.isHotspot.isComplete &&
+            this.isHotspot.result(mutation));
+
+        const cbioportalCount:boolean =
+            (this.mutationAnnotationSettings.cbioportalCount &&
+            this.getCBioportalCount.isComplete &&
+            this.getCBioportalCount.result!(mutation) >=
+            this.mutationAnnotationSettings.cbioportalCountThreshold);
+
+        const cosmicCount:boolean =
+            (this.mutationAnnotationSettings.cosmicCount &&
+            this.getCosmicCount.isComplete &&
+            this.getCosmicCount.result!(mutation) >= this.mutationAnnotationSettings.cosmicCountThreshold);
+
+        return {
+            oncoKb,
+            hotspots,
+            cbioportalCount,
+            cosmicCount
+        };
+    }
+
+    readonly geneToMutData = remoteData<{[hugoGeneSymbol:string]: Mutation[]}>({
+        await: () => [
+            this.mutations,
+            this.genes
+        ],
+        invoke: async () => {
+            const groupedByGene = _.groupBy(this.mutations.result!, (mutation: Mutation) => mutation.gene.hugoGeneSymbol);
+            return _.reduce(this.genes.result!, (memo, gene: Gene) => {
+                // a gene may have no mutations
+                memo[gene.hugoGeneSymbol] = groupedByGene[gene.hugoGeneSymbol] || [];
+                return memo;
+            },{} as { [hugoGeneSymbol:string] : Mutation[] });
+        },
+        onResult: (result:{[hugoGeneSymbol: string]: Mutation[]})=>{
+            for (const gene of Object.keys(result)) {
+                this.mutationDataCache.addData([result[gene]]);
+            }
+        }
+    });
+
+    // Mutation annotation
```

Comment:
I think at some point we should reorganize our store structure. For example, we can have a `MutationAnnotationStore` for this kind of shared data. We can then pass it to both mutation mapper store and oncoprint store.

Reply:
im open to that